### PR TITLE
Autocomplete nextcloud groups #41

### PR DESCRIPTION
**Pull request by myrho**
Autocomplete nextcloud groups in recipient fields

### CODE REVIEW

File: lib/Service/GroupsIntegration.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		$result = $this->groupManager->search($term);
+		$receivers = [];
+		foreach ($result as $g) {
+				$receivers[] = [
+					'id' => $g->getGID(),
+					'label' => $g->getDisplayName(),
+					'value' => $g->getDisplayName(),
+					'photo' => null,
+				];
+    }
```
Comment (ChristophWurst) [inline lib/Service/GroupsIntegration.php]: indentation is off

File: tests/Service/Autocompletion/AutoCompleteServiceTest.php [hunk 2]
```diff
@@ -37,11 +39,15 @@ protected function setUp() {
...
 			->getMock();
+		$this->groupsIntegration = $this->getMockBuilder('\OCA\Mail\Service\GroupsIntegration')
+			->disableOriginalConstructor()
+			->getMock();
 		$this->addressCollector = $this->getMockBuilder('\OCA\Mail\Service\AutoCompletion\AddressCollector')
 			->disableOriginalConstructor()
 			->getMock();
 
 		$this->service = new AutoCompleteService($this->contactsIntegration,
+      $this->groupsIntegration,
```
Comment (ChristophWurst) [inline tests/Service/Autocompletion/AutoCompleteServiceTest.php]: indentation

File: tests/Service/Autocompletion/AutoCompleteServiceTest.php [hunk 3]
```diff
@@ -61,10 +67,18 @@ public function testFindMatches() {
 			$john,
 		];
 
+    $groupsResult = [
```
Comment (ChristophWurst) [inline tests/Service/Autocompletion/AutoCompleteServiceTest.php]: indentation

File: tests/Service/GroupsIntegrationTest.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			->method('getDisplayName')
+			->will($this->returnValue($name));
+		return $mockGroup;
+	}
+
+
+	public function testGetMatchingGroups() {
+		$term = 'te'; // searching for: John Doe
+		$searchResult = [
+      $this->createTestGroup('testgroup', 'first test group'),
```
Comment (ChristophWurst) [inline tests/Service/GroupsIntegrationTest.php]: indentation

File: tests/Service/GroupsIntegrationTest.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      $this->createTestGroup('testgroup2', 'second test group'),
+		];
+
+		$this->groupsManager->expects($this->once())
+			->method('search')
+			->with($term)
+			->will($this->returnValue($searchResult));
+
+		$expected = [
```
Comment (ChristophWurst) [inline tests/Service/GroupsIntegrationTest.php]: indentation

**Comment by myrho** [comment]
fixed indentation and file permissions via forced push (hope that worked).

**Pull request closed without merging**